server/src/main/java/org/elasticsearch/common/geo/GeoBoundingBox.java

```diff
@@ -10,107 +10,52 @@
-import org.elasticsearch.xcontent.ToXContentFragment;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentParser;
 
 import java.io.IOException;
-import java.text.ParseException;
-import java.util.Objects;
 
 /**
  * A class representing a Geo-Bounding-Box for use by Geo queries and aggregations
  * that deal with extents/rectangles representing rectangular areas of interest.
  */
-public class GeoBoundingBox implements ToXContentFragment, Writeable {
-    static final ParseField TOP_RIGHT_FIELD = new ParseField("top_right");
-    static final ParseField BOTTOM_LEFT_FIELD = new ParseField("bottom_left");
-    static final ParseField TOP_FIELD = new ParseField("top");
-    static final ParseField BOTTOM_FIELD = new ParseField("bottom");
-    static final ParseField LEFT_FIELD = new ParseField("left");
-    static final ParseField RIGHT_FIELD = new ParseField("right");
-    static final ParseField WKT_FIELD = new ParseField("wkt");
-    public static final ParseField BOUNDS_FIELD = new ParseField("bounds");
-    public static final ParseField LAT_FIELD = new ParseField("lat");
-    public static final ParseField LON_FIELD = new ParseField("lon");
-    public static final ParseField TOP_LEFT_FIELD = new ParseField("top_left");
-    public static final ParseField BOTTOM_RIGHT_FIELD = new ParseField("bottom_right");
-
-    private final GeoPoint topLeft;
-    private final GeoPoint bottomRight;
+public class GeoBoundingBox extends BoundingBox<GeoPoint> {
 
     public GeoBoundingBox(GeoPoint topLeft, GeoPoint bottomRight) {
-        this.topLeft = topLeft;
-        this.bottomRight = bottomRight;
+        super(topLeft, bottomRight);
     }
 
     public GeoBoundingBox(StreamInput input) throws IOException {
-        this.topLeft = input.readGeoPoint();
-        this.bottomRight = input.readGeoPoint();
-    }
-
-    public boolean isUnbounded() {
-        return Double.isNaN(topLeft.lon())
-            || Double.isNaN(topLeft.lat())
-            || Double.isNaN(bottomRight.lon())
-            || Double.isNaN(bottomRight.lat());
+        super(input.readGeoPoint(), input.readGeoPoint());
     }
 
+    @Override
     public GeoPoint topLeft() {
         return topLeft;
     }
 
+    @Override
     public GeoPoint bottomRight() {
         return bottomRight;
     }
 
-    public double top() {
-        return topLeft.lat();
-    }
-
-    public double bottom() {
-        return bottomRight.lat();
-    }
-
-    public double left() {
-        return topLeft.lon();
-    }
-
-    public double right() {
-        return bottomRight.lon();
-    }
-
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        toXContentFragment(builder, true);
-        builder.endObject();
-        return builder;
-    }
-
     public XContentBuilder toXContentFragment(XContentBuilder builder, boolean buildLatLonFields) throws IOException {
```

Comment:
The value of buildLatLonFields is true almost always, and is also something geo-specific. We should get rid of it and have another method for the false case that only exists in the geo-implementation.